File: redsheet-portal/src/app/database-models/v-culture-plan.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vCulturePlan {
    constructor(vCulturePlan?: vCulturePlan )  {
        let properties = [ "culturePlanCategoryId", "culturePlanId", "culturePlanValue",];
        for (let property of properties)
            if (vCulturePlan && vCulturePlan[property])
                this[property] = vCulturePlan[property];
    }
 
    @range(1,2147483647)
	culturePlanCategoryId : number =   undefined;
 
	culturePlanId : number =   0 ;
 
    @required()
    @maxLength(1000)
	culturePlanValue : string =   undefined;


}

File: RedSheet.DbEntities/Models/vCulturePlan.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vCulturePlans", Schema = "dbo")]
	[Serializable]
	public partial class vCulturePlan  
	{
 
	[Range(1, int.MaxValue)]
	public int CulturePlanCategoryId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int CulturePlanId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string CulturePlanValue { get; set; } 

	public vCulturePlan()
	{
    }

    }
}

